compiler/hash-typecheck/src/exhaustiveness/lower.rs

```diff
@@ -108,15 +109,22 @@ impl<'tc> LowerPatOps<'tc> {
 
                 (DeconstructedCtor::Single, scope_members)
             }
+            // Since the type is already resolved, we just need to traverse down to
+            // the `Pat::Const`, and then we perform the lowering based on the type.
+            Pat::Access(AccessPat { subject, .. }) => return self.deconstruct_pat(ty, subject),
             // This is essentially a simplification to some unit nominal definition like
             // for example `None`..., here we need to be able to get the `type` of the
             // actual pattern in order to figure out which
-            Pat::Access(_) | Pat::Const(_) => {
+            Pat::Const(_) => {
                 // @@EnumToUnion: when enums aren't a thing, do this with a union and create a
                 // `DeconstructedCtor::Variant(idx)` where `idx` is the union member number
-                unreachable!()
+
+                (DeconstructedCtor::Wildcard, vec![])
```

Comment:
I don't understand why this is a wildcard? it should be a constructor..

Reply:
It depends on the type, if the underlying type is an enum variant, it's then it becomes a `DeconstructedCtor:Variant`, however in other cases `Pat::Const`as we know is used as a label for variables that have been resolved and are in scope. In the case of #461, `a` is already in scope and so becomes `Pat::Const`, but we don't care about that here and so we just treat it as a wildcard.

Reply:
Surely the constant will always point to either `Variant` or `Single`, there isn't any other valid variable that can be used in pattern position.. So why would it be a wildcard? Only binds should be treated as wildcards as I see it

Reply:
As I explained, this doesn't happen for variables that are already in scope, so for this example:
```
a := 3;
a := a; // <--- `a` is already in scope and thus becomes a `Pat::Const`
```
Since we don't care that it is a `Pat::Const` in this particular example, we should treat it as a wildcard.

Reply:
well it just depends on the semantics of this. how do you decide if LHS is pattern matching a constructor literal or is a binding/redeclaration? like what if a was an enum variant for example, instead of 3?

Reply:
the only other thing it could be is an enum variant that has no fields, which will then be handled when we switch to using unions

Reply:
Well, in that case what is the difference between bind and const? It seems to me that Pat::const is redundant

Reply:
Opened #469 to address this.